### PR TITLE
Bugfix/schema connecting cases and judges

### PR DESCRIPTION
**Pull request by jenna-anderson**
## Description

Judge names were not showing on the case table or in case details and if a user saved a case or judge it would cause an error in the backend and an infinite loading wheel, resolved only by restarting the server. 
Model functions that joined the cases to judges table were causing this backend error (see screenshot below) because judge_id no longer existed as a column in the cases table. I added judge_id back into cases as a foreign key so these joins worked properly again and I changed the findAll() model function in caseModel.js from joining judges_to_case to case and separately querying judges to directly join cases and judges.
image

judge_id was added to the case table seeds (I randomly assigned judges because this is just test seed data) and the readme documentation is updated to reflect this change in case schema.

[Corresponding frontend PR](https://github.com/Lambda-School-Labs/human-rights-first-asylum-fe-a/pull/360)

[Trello Card
](https://trello.com/c/vgdJxyz4)

[Loom Video](https://www.loom.com/share/adf64bb07afa4107a852794a70733832?sharedAppSource=personal_library)

Fixes judge names not appearing and the infinite loading wheel when saving a case or judge as a user.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by chrisjs87 — APPROVED** [review]
Great job here! I could not find anything to nitpick. Your attention to detail with adding the judge_id to the readme documentation is a really nice touch, well done!

I would check with Paul though to see why the PR page is giving a warning regarding the Sonar Cloud warning at the bottom of the page.

**Review by schandrupatla — APPROVED** [review]
Good work! your video is very detailed in explaining why we don't need judges_to_case table. The changes clean

**Review by joshuascan — APPROVED** [review]
Good job on this! Your explanation was clear and concise, making it easy to follow along and understand specifically how you were addressing this issue. It's also great that you updated the readme. It will be nice to have this functionality working again on the site.

**Comment by jenna-anderson** [comment]
> I would check with Paul though to see why the PR page is giving a warning regarding the Sonar Cloud warning at the bottom of the page.

I believe the failed check has to do with the ds_case table that is commented out. We were going to discuss if this table is needed anymore as a separate issue. Thanks for reminding me, I just added a ticket to the Trello board so we don't forget!

File: README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ The base technologies are JavaScript, HTML and CSS. The frontend leverages [Reac
     {
         "case_id": "2ff54195-ce30-456c-be63-2a6c765bdce2",
         "user_id": "00ulzdrizE2yzxToH5d6",
+        "judge_id": 2,
```
Comment (paulstgermain) [inline README.md:41]: Nice catch, making sure to update the README according to the changes made here!

File: api/cases/caseModel.js [hunk 2]
```diff
@@ -45,18 +45,11 @@ const updateCaseStatusTest = (case_id) => {
...
-      res.judges = judges.filter((judge) => judge.case_id === res.case_id);
-      return res;
-    });
-    return data;
-  });
+  const cases = db('cases as c')
+    .where({ status: 'Approved' })
+    .join('judges as j', 'j.judge_id', 'c.judge_id')
+    .select('c.*', 'j.first_name', 'j.middle_initial', 'j.last_name');
+  return cases;
```
Comment (paulstgermain) [inline api/cases/caseModel.js:52]: This is a major simplification over the prior code, and important as we aren't returning entire panels anymore - nice work.

**Review by paulstgermain — APPROVED** [review]

**Pull request merged**